packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php

```diff
@@ -199,6 +199,24 @@ public function getProductsOnCurrentDomain(int $limit, int $offset, string $orde
         return $query->execute();
     }
 
+    /**
+     * @return int
+     */
+    public function getProductsCountOnCurrentDomain(): int
+    {
+        $queryBuilder = $this->productRepository->getAllListableQueryBuilder(
+            $this->domain->getId(),
+            $this->currentCustomerUser->getPricingGroup()
+        );
+
+        $this->productRepository->addTranslation($queryBuilder, $this->domain->getLocale());
```

Comment:
Why do we need translation here?

Reply:
798a42377d9d1ac2c134986715dad926fb6c35e6